source/grid/reference_cell.cc

```diff
@@ -181,6 +181,57 @@ namespace ReferenceCell
     return Quadrature<dim>(); // never reached
   }
 
+  template <int dim>
+  Quadrature<dim> &
+  get_nodal_type_quadrature(const Type &reference_cell)
+  {
+    AssertDimension(dim, get_dimension(reference_cell));
+
+    const auto process = [&](Quadrature<dim> &quadrature) -> Quadrature<dim> & {
+      if (quadrature.size() == 0)
+        {
+          Triangulation<dim> tria;
+          make_triangulation(reference_cell, tria);
+
+          std::vector<Point<dim>> points(tria.n_vertices());
+
+          for (unsigned int v = 0; v < points.size(); ++v)
+            points[v] = tria.get_vertices()[v];
+
+          quadrature = Quadrature<dim>(points);
+        }
+
+      return quadrature;
+    };
+
+    if (reference_cell == get_hypercube(dim))
+      {
+        static Quadrature<dim> quadrature;
+        return process(quadrature);
```

Comment:
This is not thread-safe. The initialization of static variables is guaranteed to be thread-safe, but not later accesses. You will want to write this as
```
  static const Quadrature<dim> quadrature = ...;
  return quadrature;
```

Reply:
I think a more obvious design would also pass the desired reference cell to the `process` function (which I might rename `create_quadrature(const ReferenceCell::Type &)` since the function does something different depending on what argument the *outer* function is being called with.